internal/repository/task.go

```diff
@@ -0,0 +1,132 @@
+package repository
+
+import (
+	"database/sql"
+	"fmt"
+	"go_final_project/config"
+	"go_final_project/internal/models"
+	"log"
+	"strconv"
+	"time"
+)
+
+func AddTaskToDB(task *models.Task) (int64, error) {
+	query := "INSERT INTO scheduler (date, title, comment, repeat) VALUES (?, ?, ?, ?)"
+	res, err := config.GetDB().Exec(query, task.Date, task.Title, task.Comment, task.Repeat)
+	if err != nil {
+		return 0, err
+	}
+	return res.LastInsertId()
+}
+
+func UpdateTaskToDB(task *models.Task) error {
+	query := "UPDATE scheduler SET date = ?, title = ?, comment = ?, repeat = ? WHERE id = ?"
+	_, err := config.GetDB().Exec(query, task.Date, task.Title, task.Comment, task.Repeat, task.Id)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func UpdateDateToDB(date string, task models.Task) error {
+	query := "UPDATE scheduler SET date = ? WHERE id = ?"
+	_, err := config.GetDB().Exec(query, date, task.Id)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func GetTaskById(strId string) (models.Task, error) {
+	id, err := strconv.Atoi(strId)
+	if err != nil {
+		log.Printf("Ошибка преобразования id: %v", err)
+		return models.Task{}, fmt.Errorf("некорректный id: %w", err)
+	}
+
+	db := config.GetDB()
+	task := models.Task{}
+
+	query := "SELECT id, date, title, comment, repeat FROM scheduler WHERE id = ?"
+
+	// Выполняем запрос
+	row := db.QueryRow(query, id)
+	err = row.Scan(&task.Id, &task.Date, &task.Title, &task.Comment, &task.Repeat)
+	if err != nil {
+		if err == sql.ErrNoRows {
```

Comment:
Лучше так проверять ошибки, так как при обертывании ошибки простое сравнение уже не сработает
```suggestion
		if errors.Is(err, sql.ErrNoRows) {
```